From fb35a30fa077d2cf8202204a1f6a6ffb5ad0663b Mon Sep 17 00:00:00 2001
From: Danny Nold <dannynold@freescale.com>
Date: Sun, 7 Aug 2011 21:30:35 -0500
Subject: [PATCH 0324/2508] ENGR00154431 - MXCFB_SET_WAVEFORMS ioctl broken

- Fixed bug in how new waveform set is copied into EPDC driver internal
copy of waveform modes.

Signed-off-by: Danny Nold <dannynold@freescale.com>
---
 drivers/video/mxc/mxc_epdc_fb.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/video/mxc/mxc_epdc_fb.c b/drivers/video/mxc/mxc_epdc_fb.c
index 44414ed..2241689 100644
--- a/drivers/video/mxc/mxc_epdc_fb.c
+++ b/drivers/video/mxc/mxc_epdc_fb.c
@@ -1428,7 +1428,7 @@ void mxc_epdc_fb_set_waveform_modes(struct mxcfb_waveform_modes *modes,
 	struct mxc_epdc_fb_data *fb_data = info ?
 		(struct mxc_epdc_fb_data *)info:g_fb_data;
 
-	memcpy(&fb_data->wv_modes, modes, sizeof(modes));
+	memcpy(&fb_data->wv_modes, modes, sizeof(struct mxcfb_waveform_modes));
 }
 EXPORT_SYMBOL(mxc_epdc_fb_set_waveform_modes);
 
-- 
1.7.7.4

